### PR TITLE
[11.x] Fix `getDefaultUserProvider` docblock

### PR DESCRIPTION
**Pull request by daniser**
Currently, creating user provider using `createUserProvider` without explicitly passing its name will always return `null` unless `auth.defaults.provider` config entry is defined (which is [not defined out-of-the-box](https://github.com/laravel/laravel/blob/b378ce946a05b5ab80776c3b5e40fa84751084bd/config/auth.php#L16-L19) btw).

The reason is, `getDefaultUserProvider` method - contrary to its name and typehint - returns `null` when said config entry is missing.

This PR changes that to fall back to default [`'users'` provider](https://github.com/laravel/laravel/blob/b378ce946a05b5ab80776c3b5e40fa84751084bd/config/auth.php#L63) instead of `null`.

### CODE REVIEW

File: src/Illuminate/Auth/CreatesUserProviders.php [hunk 1]
```diff
@@ -48,11 +48,11 @@ public function createUserProvider($provider = null)
      * @param  string|null  $provider
      * @return array|null
```
Comment (rodrigopedra) [inline src/Illuminate/Auth/CreatesUserProviders.php:49]: It seems the return value can't be `null` anymore, right?

**Comment by rodrigopedra** [comment]
Not sure, I follow the reasoning for this change. 

Maybe the `auth.defaults.provider` key is left blank intentionally, and thus no provider should be returned.

What are the benefits of having an out-of-the-box user provider? Granted that each shipped guard defines their own provider.

What issues does this change solve?

Also, defaulting to some hard-coded value doesn't seem intuitive to me, if someone does not expect it to return anything, hunting for this value could be cumbersome.

As this is also a change in default behavior of a public method, maybe targeting 12.x and adding a note to the upgrade guide is a better idea.

File: src/Illuminate/Auth/CreatesUserProviders.php (hunk 1 above)
Comment (daniser) [reply src/Illuminate/Auth/CreatesUserProviders.php:49]: Not exactly. It will return `null` if given provider doesn't exist.

**Comment by daniser** [comment]
> Maybe the auth.defaults.provider key is left blank intentionally, and thus no provider should be returned.

I found original PR (#18856) and I think your point is valid.

My use case was to use `CreatesUserProviders` trait outside auth subsystem as thin abstraction layer over Eloquent to find users by their credentials. Maybe that's not intended use case, but anyway, anyone who want to define default user provider may override `getDefaultUserProvider` method.

I've reverted my changes and instead fixed `getDefaultUserProvider` docblock to reflect that `null` is allowed return value.

I also changed `GuardHelpers::setProvider` method so it is now chainable.

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If applicable, please consider releasing your code as a package so that the community can still take advantage of your contributions!

**Pull request closed without merging**